perception/navigator_vision/nodes/pinger_finder.py

```diff
@@ -0,0 +1,170 @@
+
+
+# receives pings and adds the representation of a line between the hydrophone array and the pinger
+# to our line_arr linear equation system
+def ping_cb(processed_ping):
+    global line_array, sample_amplitudes
+    print "Got processed ping message:\n{}".format(processed_ping)
+    if abs(processed_ping.freq - target_freq) < freq_tol and processed_ping.amplitude > min_amp:
+        print "\x1b[34mTrustworthy pinger heading\x1b[0m"
+        trans, rot = None, None
+        try:
+            tf_listener.waitForTransform("/enu", "/hydrophones", processed_ping.header.stamp, rospy.Duration(0.25))
+            trans, rot = tf_listener.lookupTransform("/enu", "/hydrophones", processed_ping.header.stamp)
+            p0 = np.array([trans[0], trans[1]])
+            R = tf.transformations.quaternion_matrix(rot)[:3, :3]
+            print "quat: {} rot mat: {}".format(rot, R)
+            print "delta hyd frame: {}".format(processed_ping.position)
+            delta = R.dot(navigator_tools.point_to_numpy(processed_ping.position))[:2]
+            print "delta enu frame: {}".format(delta)
+            p1 = p0 + point_distance * delta / npl.norm(delta)
+            line_coeffs = np.array([[p0[0], p0[1], p1[0], p1[1]]]) # p0 and p1 define a line
+            visualize_line(Point(p0[0], p0[1], 0), Point(p1[0], p1[1], 0))
+            if listen:
+               line_array = np.append(line_array, line_coeffs, 0)
+               sample_amplitudes = np.append(sample_amplitudes, processed_ping.amplitude)
+            else:
+                print "Not currently listening"
+        except (tf.LookupException, tf.ConnectivityException, tf.ExtrapolationException) as e:
+            print e
+        # delete softest samples if we have over max_samples
+        if len(line_array) >= max_samples:
+            softest_idx = np.argmin(sample_amplitudes)
+            line_array = np.delete(line_array, softest_idx, axis=0)  
+            sample_amplitudes = np.delete(sample_amplitudes, softest_idx)
+        print "Number of good samples: {}".format(line_array.shape)
+    else:
+        print "Untrustworthy pinger heading. Freq = {} kHZ".format(processed_ping.freq)
+
+def visualize_pinger(event):
+    marker = Marker()
+    marker.ns = "pinger"
+    marker.header.stamp = rospy.Time(0)
+    marker.header.frame_id = "/enu"
+    marker.type = marker.CUBE
+    marker.action = marker.ADD
+    marker.scale.x = 1.0
+    marker.scale.y = 1.0
+    marker.scale.z = 1.0
+    marker.color.g = 1.0
+    marker.color.a = 1.0
+    marker.pose.orientation.w = 1.0
+    marker.pose.position = pinger_position
+    if pinger_position != Point(0, 0, 0):
+        print "vis pinger"
+        viz_pub.publish(marker)
+    else:
+        print "pinger position: {}".format(pinger_position)
+
+def visualize_line(p0, p1):
+    global debug_arrow_id
+    marker = Marker()
+    marker.ns = "pinger/debug"
+    marker.header.stamp = rospy.Time(0)
```

Comment:
Maybe this should be set to the current time, ```ros::Time()``` . That may be why the markers are sticking around too long.